File: windows/RNGetRandomValues/RNGetRandomValues.cpp
#include "pch.h"
#include "RNGetRandomValues.h"

namespace winrt::RNGetRandomValues::implementation
{
    REACT_METHOD(getRandomBase64)
    std::string RNGetRandomValues::getRandomBase64(int byteLength)
    {
        winrt::Windows::Storage::Streams::IBuffer buffer = winrt::Windows::Security::Cryptography::CryptographicBuffer::GenerateRandom(byteLength);
        return winrt::to_string(winrt::Windows::Security::Cryptography::CryptographicBuffer::EncodeToBase64String(buffer));
    }
}

File: windows/RNGetRandomValues/RNGetRandomValues.cs
using System;
using Microsoft.ReactNative.Managed;
using Windows.Security.Cryptography;
using Windows.Storage.Streams;

namespace RNGetRandomValuesModule
{
    [ReactModule]
    class RNGetRandomValues
    {
        [ReactSyncMethod]
        public string getRandomBase64(int byteLength)
        {
            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteLength);
            return CryptographicBuffer.EncodeToBase64String(buffer);
        }
    }
}
